conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandlerTest.java

```diff
@@ -361,6 +362,13 @@ public void handlesErrorWithoutRethrowing() {
                 .doesNotThrowAnyException();
     }
 
+    @Test
+    public void handlesPeerRemoteExceptionsWithoutRethrowing() throws IOException {
+        exception = UndertowMessages.MESSAGES.couldNotReadContentLengthData();
+        HttpURLConnection connection = execute();
+        assertThat(connection.getResponseCode()).isEqualTo(200);
```

Comment:
there shouldn't be a response code if the request wasn't fully read and we threw before processing & writing response
```suggestion
        assertThatThrownBy(() -> connection.getResponseCode())
                .isInstanceOf(SocketException.class)
                .hasMessageContaining("Unexpected end of file from server");
```